docs/operator-manual/applicationset/Generators-Git.md

```diff
@@ -382,6 +385,61 @@ spec:
 
 In `values` we can also interpolate all fields set by the git files generator as mentioned above.
 
+### Pass additional parameter files via `extraParameterFiles` field
+
+You may provide an additional list of files containing parameters (yaml or json format) with the `extraParameterFiles` of the git generator.
+Extra parameters added via the `extraParameterFiles` field are added as `extraParams.(field)`.
+
+These files are expected to be resolved in the same git repository as the rest of the generator.
+
+The files provided supports glob format, and you can use any parameter resolved from the files or directories fields to templatize the file path.
+An extra parameter file not found will not trigger an error, and will just be ignored.
+
+A hierarchy occurs in case of collision betweens the parameters, as follows:
+
+* Parameters are merged and overriden file by file, from top to bottom of the list provided
+* When a file provided is a glob pattern, files are sorted by path and overrides occur in the sorted order.
+
+This is designed to be used like the helm values provided inside an Application's helm source, but allowing to templatize the whole Application.
+
+In this example, we gather two different files:
+
+* A set of common parameters for all the file generators
+* A specific parameter set for the cluster name found inside each generator file
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: guestbook
+  namespace: argocd
+spec:
+  goTemplate: true
+  goTemplateOptions: ["missingkey=error"]
+  generators:
+  - git:
+      repoURL: https://github.com/argoproj/argo-cd.git
+      revision: HEAD
+      files:
+      - path: "applicationset/examples/git-generator-files-discovery/cluster-config/**/config.json"
+      extraParameterFiles:
+        - "mypath/common-parameters/**/*.json
+        - "myotherpath/specific-params/params-{{.cluster.name}}.yaml"
```

Comment:
```suggestion
      - path: "applicationset/examples/git-generator-files-discovery/cluster-config/**/config.json"
      - path: "mypath/common-parameters/**/*.json
        prefix: extraParams
      - path: "myotherpath/specific-params/params-{{.cluster.name}}.yaml"
        prefix: extraExtraParams
```

Thoughts?

Reply:
Nice ! It just complexifies a bit the whole thing as I was implementing a basic yaml merge as helm is doing with values files.
But we can just add the extra top level from the prefix and do the merge anyway, so it should work !

Reply:
Updated with that change